### PR TITLE
TEZ-4561: Improve reported exception when DAGAppMaster is shutting down.

### PR DESCRIPTION
**Pull request by ayushtkn**

### CODE REVIEW

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  26m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  19m  6s |  master passed  |
| +1 :green_heart: |  compile  |   0m 43s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  compile  |   0m 38s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  checkstyle  |   1m 35s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 40s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 21s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +0 :ok: |  spotbugs  |   1m 52s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 50s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 20s |  tez-dag: The patch generated 1 new + 51 unchanged - 0 fixed = 52 total (was 51)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  findbugs  |   1m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  8s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 18s |  The patch does not generate ASF License warnings.  |
|  |   |  60m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/365 |
| JIRA Issue | TEZ-4561 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 56fdc97417a9 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / cd6ceec3d |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/1/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/1/testReport/ |
| Max. process+thread count | 244 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/1/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ayushtkn** [comment]
Added a test.
Sharing the output of exception
When State is INITED
```
org.apache.tez.dag.api.TezUncheckedException: Cannot get ApplicationACLs before all services have started, The current service state is INITED
```

When State is STOPPED
```
org.apache.tez.dag.api.TezUncheckedException: Cannot get ApplicationACLs before all services have started, The current service state is STOPPED. The shutdown hook started at Fri Aug 16 16:08:09 IST 2024
```

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  17m  7s |  master passed  |
| +1 :green_heart: |  compile  |   0m 45s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  compile  |   0m 39s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  checkstyle  |   1m 43s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 21s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +0 :ok: |  spotbugs  |   1m 59s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 57s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 20s |  tez-dag: The patch generated 2 new + 57 unchanged - 3 fixed = 59 total (was 60)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  findbugs  |   1m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  7s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 18s |  The patch does not generate ASF License warnings.  |
|  |   |  33m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/365 |
| JIRA Issue | TEZ-4561 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux eb8c42aed208 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / cd6ceec3d |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/2/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/2/testReport/ |
| Max. process+thread count | 245 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/2/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  16m 24s |  master passed  |
| +1 :green_heart: |  compile  |   0m 43s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  compile  |   0m 38s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 21s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +0 :ok: |  spotbugs  |   1m 51s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 48s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 1 new + 57 unchanged - 3 fixed = 58 total (was 60)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  findbugs  |   1m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  8s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 17s |  The patch does not generate ASF License warnings.  |
|  |   |  32m 16s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/365 |
| JIRA Issue | TEZ-4561 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 06ebcb967f9f 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / cd6ceec3d |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/3/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/3/testReport/ |
| Max. process+thread count | 222 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/3/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java [hunk 1]
```diff
@@ -1743,6 +1747,13 @@ public void setQueueName(String queueName) {
     }
   }
 
+  private String getShutdownTimeString() {
+    if (shutdownHandler != null && shutdownHandler.shutdownTime != null) {
+      return ". The shutdown hook started at " + shutdownHandler.shutdownTime;
```
Comment (abstractdog) [inline tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java:1752]: It's strange that this separate string generation method takes care of the closing point of the previous sentence, I believe that should go back to  getApplicationACLs

Comment (ayushtkn) [reply tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java]: I moved the "." back to the {{getApplicationACLs}}

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  27m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  16m 59s |  master passed  |
| +1 :green_heart: |  compile  |   0m 44s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  compile  |   0m 39s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  checkstyle  |   1m 31s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 36s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 23s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +0 :ok: |  spotbugs  |   1m 53s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 51s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 20s |  tez-dag: The patch generated 1 new + 57 unchanged - 3 fixed = 58 total (was 60)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  findbugs  |   1m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  6s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 17s |  The patch does not generate ASF License warnings.  |
|  |   |  59m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/365 |
| JIRA Issue | TEZ-4561 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 249eda176fba 5.15.0-119-generic #129-Ubuntu SMP Fri Aug 2 19:25:20 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 174d4e3bb |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/4/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/4/testReport/ |
| Max. process+thread count | 205 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/4/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by abstractdog** [comment]
nit: what about changing shutdownTime to private?
not necessarily because of the checkstyle warning (private + accessor methods), but instead to keep it inline with other fields of DAGAppMasterShutdownHandler, like:
```
    private AtomicBoolean shutdownHandled = new AtomicBoolean(false);
    private long sleepTimeBeforeExit = TezConstants.TEZ_DAG_SLEEP_TIME_BEFORE_EXIT;
```
I guess the usage scope of those are similar

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  13m 50s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 41s |  master passed  |
| +1 :green_heart: |  compile  |   0m 26s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  compile  |   0m 23s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 17s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +0 :ok: |  spotbugs  |   1m 22s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 20s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 18s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javac  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 17s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  javac  |   0m 17s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  tez-dag: The patch generated 0 new + 57 unchanged - 3 fixed = 57 total (was 60)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m  8s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m  7s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| -1 :x: |  findbugs  |   0m 47s |  tez-dag generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 24s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  39m 13s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:tez-dag |
|  |  org.apache.tez.dag.app.DAGAppMaster$DAGAppMasterShutdownHandler.getShutdownTime() may expose internal representation by returning DAGAppMaster$DAGAppMasterShutdownHandler.shutdownTime  At DAGAppMaster.java:by returning DAGAppMaster$DAGAppMasterShutdownHandler.shutdownTime  At DAGAppMaster.java:[line 943] |
|  |  org.apache.tez.dag.app.DAGAppMaster$DAGAppMasterShutdownHandler.setShutdownTime(Date) may expose internal representation by storing an externally mutable object into DAGAppMaster$DAGAppMasterShutdownHandler.shutdownTime  At DAGAppMaster.java:by storing an externally mutable object into DAGAppMaster$DAGAppMasterShutdownHandler.shutdownTime  At DAGAppMaster.java:[line 947] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/365 |
| JIRA Issue | TEZ-4561 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 2ba8942efbd2 5.15.0-116-generic #126-Ubuntu SMP Mon Jul 1 10:14:24 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 174d4e3bb |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/5/artifact/out/new-findbugs-tez-dag.html |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/5/testReport/ |
| Max. process+thread count | 227 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/5/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  19m  8s |  master passed  |
| +1 :green_heart: |  compile  |   0m 45s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  compile  |   0m 40s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  checkstyle  |   1m 42s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 40s |  master passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m 23s |  master passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +0 :ok: |  spotbugs  |   2m  0s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 58s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 32s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javac  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 0 new + 57 unchanged - 3 fixed = 57 total (was 60)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~22.04-b05  |
| +1 :green_heart: |  findbugs  |   1m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  7s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 19s |  The patch does not generate ASF License warnings.  |
|  |   |  35m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/365 |
| JIRA Issue | TEZ-4561 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 125dfebe230f 5.15.0-119-generic #129-Ubuntu SMP Fri Aug 2 19:25:20 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 174d4e3bb |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu322.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~22.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/6/testReport/ |
| Max. process+thread count | 233 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-365/6/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ayushtkn** [comment]
@abstractdog I have changed the variable to private & used setter & getter, there was some FindBugs warning, which I have dodged, lemme know if things look good

**Review by abstractdog — APPROVED** [review]
+1

**Pull request merged**